pkg/util/cache/lru_cache.go

```diff
@@ -0,0 +1,164 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package cache
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/telemetry"
+	"unsafe"
+)
+
+var (
+	// There are multiple instances of the interner, one per worker. Counters are normally fine,
+	// gauges require special care to make sense. We don't need to clean up when an instance is
```

Comment:
This looks like its based on a slightly stale version of the code, see this PR for the "new" telemetry https://github.com/DataDog/datadog-agent/pull/20420